include/circt/Dialect/FIRRTL/Canonicalization.td

```diff
@@ -43,4 +43,16 @@ def GTWithConstLHS : Pat<
   (LTPrimOp $rhs, $lhs),
   [(NonConstantOp $rhs)]>;
 
+// mux(cond, x, mux(cond, y, z))
+def MuxSameCondLow : Pat<
+  (MuxPrimOp $cond, $x, (MuxPrimOp $cond, $y, $z)),
+  (MuxPrimOp $cond, $x, $z)
+  >;
+
+// mux(cond, x, node(mux(cond, y, z)))
+def MuxSameCondLowNode : Pat<
+  (MuxPrimOp $cond, $x, (NodeOp (MuxPrimOp $cond, $y, $z), $name, $attr)),
```

Comment:
Please don't include Node into this pattern.  We should separately eliminate any node operations (or drop them from the firrtl dialect entirely).